src/pkg/services/m365/m365.go

```diff
@@ -98,55 +112,56 @@ type Site struct {
 
 // Sites returns a list of Sites in a specified M365 tenant
 func Sites(ctx context.Context, acct account.Account, errs *fault.Bus) ([]*Site, error) {
-	gc, err := connector.NewGraphConnector(ctx, graph.HTTPClient(graph.NoTimeout()), acct, connector.Sites, errs)
+	sites, err := discovery.Sites(ctx, acct, errs)
 	if err != nil {
 		return nil, clues.Wrap(err, "initializing M365 graph connection")
 	}
 
-	// gc.Sites is a map with keys: SiteURL, values: ID
-	ret := make([]*Site, 0, len(gc.Sites))
-	for k, v := range gc.Sites {
-		ret = append(ret, &Site{
-			WebURL: k,
-			ID:     v,
-		})
+	ret := make([]*Site, 0, len(sites))
+
+	for _, s := range sites {
+		ps, err := parseSite(s)
+		if err != nil {
+			return nil, clues.Wrap(err, "parsing siteable")
+		}
+
+		ret = append(ret, ps)
 	}
 
 	return ret, nil
 }
 
-// SiteURLs returns a list of SharePoint site WebURLs in the specified M365 tenant
-func SiteURLs(ctx context.Context, acct account.Account, errs *fault.Bus) ([]string, error) {
-	gc, err := connector.NewGraphConnector(ctx, graph.HTTPClient(graph.NoTimeout()), acct, connector.Sites, errs)
-	if err != nil {
-		return nil, clues.Wrap(err, "initializing M365 graph connection")
+// parseSite extracts the information from `models.Siteable` we care about
+func parseSite(item models.Siteable) (*Site, error) {
+	s := &Site{
+		ID:     ptr.Val(item.GetId()),
+		WebURL: ptr.Val(item.GetWebUrl()),
 	}
 
-	return gc.GetSiteWebURLs(), nil
+	return s, nil
 }
 
-// SiteIDs returns a list of SharePoint sites IDs in the specified M365 tenant
-func SiteIDs(ctx context.Context, acct account.Account, errs *fault.Bus) ([]string, error) {
-	gc, err := connector.NewGraphConnector(ctx, graph.HTTPClient(graph.NoTimeout()), acct, connector.Sites, errs)
+// SitesMap retrieves all sites in the tenant, and returns two maps: one id-to-webURL,
+// and one webURL-to-id.
+func SitesMap(
+	ctx context.Context,
+	acct account.Account,
+	errs *fault.Bus,
+) (common.IDsNames, error) {
```

Comment:
do we intend SDK consumers to use this? If we do we may need to move `IDsNames` to a different package as it's in `internal` right now

Reply:
Since consumers shouldn't need to _populate_ idsnames, only receive and use its interface (and pass it into interface-accepting funcs, if anything), it's okay for the struct to remain internal.